kroxylicious-integration-tests/src/test/java/io/kroxylicious/kms/service/KmsIT.java

```diff
@@ -24,25 +25,36 @@
  * @param <E> The type of encrypted DEK
  */
 @ExtendWith(TestKmsFacadeInvocationContextProvider.class)
-class KmsIT<C, K, E> {
+class KmsIT<C, I, K, E> {
 
     private Kms<K, E> kms;
 
     private TestKekManager manager;
     private String alias;
     private K resolvedKekId;
+    private KmsService<C, I, K, E> service;
+    private I initData;
 
     @BeforeEach
     @SuppressWarnings("unchecked")
-    void beforeEach(TestKmsFacade<?, ?, ?> facade) throws Exception {
-        var service = (KmsService<C, K, E>) facade.getKmsServiceClass().getDeclaredConstructor(new Class[]{}).newInstance();
-        kms = service.buildKms((C) facade.getKmsServiceConfig());
+    void beforeEach(TestKmsFacade<?, ?, ?, ?> facade) throws Exception {
+        service = (KmsService<C, I, K, E>) facade.getKmsServiceClass().getDeclaredConstructor(new Class[]{}).newInstance();
+        C kmsServiceConfig = (C) facade.getKmsServiceConfig();
+        initData = service.initialize(kmsServiceConfig);
+        kms = service.buildKms(initData);
         manager = facade.getTestKekManager();
 
         alias = "alias-" + UUID.randomUUID();
         resolvedKekId = createAndResolve(alias);
     }
 
+    @AfterEach
+    void afterEach() {
+        if (service != null && initData != null) {
```

Comment:
Hmm maybe we should document the initData as being `nullable` so we don't need the second guard condition.